### DISCUSSION TITLE
Jobs queued with Model Id 0

### DISCUSSION THREAD

**Discussion by jasontxf** [laravel/framework#45521] [Q&A] [answered]
- Horizon Version: 5.10.5
- Laravel Version: 9.43.0
- PHP Version: 8.1.5
- Redis Driver & Version: predis/phpredis (How do I check in forge? Not in composer.json) Redis version: 5.0.7
- Database Driver & Version: mysql 8.0.21

### Description:
When queueing jobs with a model, the queued job model id becomes 0 which resulted in the actual queued job not being able to be processed.

![Xnapper-2023-01-05-11 48 12](https://user-images.githubusercontent.com/1124731/210697664-b9b35a67-a704-496b-8ff3-0d9fef17360c.png)

It doesn't happen all the time, but it happened frequently enough to become a problem for me. This happened right after I upgraded Laravel to 9.43.0. Previously it was working 100% of the time

### Steps To Reproduce:

I merely queue it the normal way.

```php
app(SendTelegramMessage::class)
    ->dispatch($log);
```
I have no idea how to troubleshoot further as it's not happening all the time. 
Any help/clues will be greatly appreciated. Thanks!

**Comment by ankurk91** [comment]
What is the schema for `App\MessageLog` table?
How are you dispatching this job?

Can you share code your for this job? Specially constructor

**Comment by jasontxf** [comment]
Hello, 

Thanks for replying!

Here's the Schema for MessageLog Table
```php
Schema::create('message_logs', function (Blueprint $table) {
			$table->id();
			$table->foreignId('user_id')->nullable()->constrained()->onDelete('cascade');
			$table->string('channel')->index(); 
			$table->string('source')->index(); 
			$table->text('message')->nullable();
			$table->text('referrer')->nullable();
			$table->json('payload')->nullable();
			$table->json('headers')->nullable();
			$table->timestamps();
		});
```

Just a very typical model.

I dispatched the job using the following code as stated above

```php
app(SendTelegramMessage::class)
    ->dispatch($log);
```

The constructor for the jobs looks like this

```php
<?php

namespace App\Jobs;

class SendTelegramMessage implements ShouldQueue
{
    use Dispatchable;
    use InteractsWithQueue;
    use Queueable;
    use SerializesModels;

    /**
     * @var TelegramChat
     */
    public $log;
    
    /**
     * Create a new job instance.
     *
     * @param  MessageLog  $log
     */
    public function __construct(MessageLog $log)
    {
        $this->queue = 'message';
        
        $this->log = $log;
    }

}
```

Hope it helps!

**Reply by henzeb** [reply]
Did you save the model first?

**Reply by jasontxf** [reply]
Hi, 

Yeah, I did. Only after the model is created, then i dispatched it. Here's the code for the send function

```php
  public function send($message, array $options = null)
  {
      $log = app(MessageLog::class)
          ->createTelegramLog($message, $this, $options);

      app(SendTelegramMessage::class)
          ->dispatch($log);
    }
```

and within the `@createTelegramLog` method

```php
   public function createTelegramLog($text, TelegramChat $chat, $options = [])
    {
        return self::create([...]);
    }
```

This issue happened every now and then which is why it really puzzles me so much. 
Thanks for taking a look!

**Reply by ankurk91** [reply]
Can you try this

```php
app(SendTelegramMessage::class)
          ->dispatch($log->refresh());
```

**Reply by jasontxf** [reply]
That's a good idea! I'll give this a try and update again. Thanks!

**Reply by jasontxf** [reply]
Sadly, it's still happening :(

Really have no clue why is this happening.

**Comment by lioneaglesolutions** [comment]
I am having the same issue... no idea what the cause is. It seems to happen randomly. @jasontxf any luck fixing it?

**Reply by jasontxf** [reply]
Hello,

No luck fixing it :( Though one thing I'm sure, this only happened right after I upgrade to Laravel 9. So it should still be something to do with the framework. Just not sure what...

**Reply by lioneaglesolutions** [reply]
Very strange - I have this pop up every now and again and can not seem to work it out! Going to try just queuing with the model ID instead...

**Reply by jasontxf** [reply]
Same with me! It doesn't happen all the time. But it happens frequent enough for me to be worrying. Please keep me updated if you manage to fix this issue. I'm still trying to figure out the issue on my part here. We are on our own since no one is experiencing this :(

**Reply by lioneaglesolutions** [reply]
So I just had a look at the `failed_jobs` table and the jobs that fail this way aren't even in there... but all of the other ones are

**Reply by jasontxf** [reply]
Yes, same with me!
I've made sure to remove $deleteWhenMissingModels so that I could see the failed jobs. But I cant see it too :( 

This is getting so disappointing... seems like no one else has this issue except us.

**Reply by lioneaglesolutions** [reply]
Well the other thing is that this is actually a slightly different job now... same issue. The jobs are dispatched from webhooks, in this case from Mailgun.

1. Receive webhook
2. Dispatch job

The job has a transaction in there but that shouldn't affect the serialisation of the model.

**Reply by lioneaglesolutions** [reply]
Seems to happen if webhooks are received in quick succession but not always.

Any similarities there?

**Reply by jasontxf** [reply]
It happened randomly but often enough. So when you are receiving many webhooks simultaneously, the chances of it happening are higher. So you will be able to see it happen.

**Reply by lioneaglesolutions** [reply]
@jasontxf are you using Telescope?

I found this - https://github.com/laravel/telescope/issues/289

**Reply by jasontxf** [reply, marked as answer]
oh wow! How did I miss such an important issue. 
I didn't think Telescope could be the one causing it!

Thanks for sharing.

I am indeed using Telescope.
I'm going to try their solution one by one and see if any works for me.
1. Review Telescope Config File
2. Separate DB connection to another one
3. Disable Query Watcher

Today I will reviewed telescope config file and updated be the same as the latest version.
If it doesn't work I'll move to step 2 and eventually step 3. 
Cross-finger one of the solutions will work! 

@lioneaglesolutions have you tried any of the solution? Did it work?

**Reply by lioneaglesolutions** [reply]
Trying #3 first and then will probably go to number 2 anyway

**Reply by jasontxf** [reply]
Ok! Let's keep each other updated on what works :) Thanks!

**Reply by lioneaglesolutions** [reply]
So #3 did nothing - issue still occurring. Going to try a different Telescope connection

**Reply by jasontxf** [reply]
@lioneaglesolutions Just an update, my issue didn't seems to occur for the entire day.
Going to monitor for one whole day before I make any declaration :)

**Reply by lioneaglesolutions** [reply]
I have disabled Telescope completely and the issue seems to be resolved.

**Reply by jasontxf** [reply]
Hello, I only tried #1 so far and i dont seems to be facing anymore issues. Makes me wonder if my queue is working haha. So I think it's safe to say Telescope is definitely the issue for both of us. Just that we have no idea which part.

I'm going to further monitor for an entire week and see if the issue pops up.

**Reply by lioneaglesolutions** [reply]
I mean it's super strange that this doesn't happen to anyone else (?) seemingly... be good if I could somehow easily reproduce.

**Reply by jasontxf** [reply]
yeah I agree... I have 3 Laravel apps that is running in production and now i'm afraid to upgrade in the event i encountered the same issue again. Especially when those app have hundreds of jobs to process of different type. 

Facing this issue would be catastrophe for me.

**Reply by jasontxf** [reply]
So did you try all #1 #2 and #3 before you disable Telescope?

**Reply by lioneaglesolutions** [reply]
I didn't try number 2 yet - just went straight to 3
